Typecheck error on component using `generic` and `defineModel` with a custom name containing a dash

### Vue - Official extension or vue-tsc version

2.1.6

### VSCode version

Not relevant

### Vue version

3.5.8

### TypeScript version

5.6.2

### System Info

```shell
Not relevant
```


### package.json dependencies

_No response_

### Steps to reproduce

https://stackblitz.com/edit/vitejs-vite-nctods?file=src%2FApp.vue

`App.vue` defines a `generic` as well as uses `defineModel` with a custom name containing dash.
Run `pnpm test` to type-check the project, and the weird error will occur:

```
App.vue:1:34 - error TS2304: Cannot find name 'A'.

1 <script setup lang="ts" generic="A">
                                   ~
```

A grammar error occurs if multiple generic parameters are present:

```
App.vue:1:35 - error TS1005: '>' expected.

1 <script setup lang="ts" generic="A, B">
                                    ~
```

Try to remove a dash from `hello-world` and see the both errors go away.

### What is expected?

No such errors

### What is actually happening?

They do occur

### Link to minimal reproduction

_No response_

### Any additional comments?

_No response_

--- a/packages/language-core/lib/codegen/script/scriptSetup.ts
+++ b/packages/language-core/lib/codegen/script/scriptSetup.ts
@@ -438,7 +438,7 @@ function* generateModelEmits(
 		const generateDefineModels = function* () {
 			for (const defineModel of defineModels) {
 				const [propName, localName] = getPropAndLocalName(scriptSetup, defineModel);
-				yield `'update:${propName}': [${propName}:`;
+				yield `'update:${propName}': [value: `;
 				yield* generateDefinePropType(scriptSetup, propName, localName, defineModel);
 				yield `]${endOfLine}`;
 			}

--- a/test-workspace/tsc/passedFixtures/vue3/#4890/main.vue
+++ b/test-workspace/tsc/passedFixtures/vue3/#4890/main.vue
@@ -1,0 +1,3 @@
+<script setup lang="ts" generic="T">
+defineModel('hello-world');
+</script>

